server/lib/report_server_web/live/codap_plugin_live/index.ex

```diff
@@ -0,0 +1,90 @@
+defmodule ReportServerWeb.CodapPluginLive.Query do
+  defstruct name: nil, sql: nil
+
+  def count_sql(table, name, period, count \\ "*")
+  def count_sql(table, name, :month, count), do: "SELECT DATE_FORMAT(created_at, '%Y-%m') AS date, COUNT(#{count}) AS num_#{name} from #{table} GROUP BY YEAR(created_at), MONTH(created_at)"
+  def count_sql(table, name, :year, count), do: "SELECT YEAR(created_at) AS date, COUNT(#{count}) AS num_#{name} from #{table} GROUP BY YEAR(created_at)"
+  def user_count_sql(table, name, :month), do: count_sql(table, name, :month, "DISTINCT user_id")
+  def user_count_sql(table, name, :year), do: count_sql(table, name, :year, "DISTINCT user_id")
+end
+
+defmodule ReportServerWeb.CodapPluginLive.Index do
+  use ReportServerWeb, :live_view
+
+  alias ReportServer.PortalDbs
+  alias ReportServerWeb.CodapPluginLive.Query
+
+  @servers ["learn.concord.org", "ngss-assessment.portal.concord.org"] #, "learn.portal.staging.concord.org"]
+  @queries %{
+    "new_student_counts_by_month" => %Query{name: "New Student Counts By Month", sql: Query.user_count_sql("portal_students", "students", :month)},
+    "new_student_counts_by_year" => %Query{name: "New Student Counts By Year", sql: Query.user_count_sql("portal_students", "students", :year)},
+    "new_teacher_counts_by_month" => %Query{name: "New Teacher Counts By Month", sql: Query.user_count_sql("portal_teachers", "teachers", :month)},
+    "new_teacher_counts_by_year" => %Query{name: "New Teacher Counts By Year", sql: Query.user_count_sql("portal_teachers", "teachers", :year)},
+    "new_class_counts_by_month" => %Query{name: "New Class Counts By Month", sql: Query.count_sql("portal_clazzes", "classes", :month)},
+    "new_class_counts_by_year" => %Query{name: "New Class Counts By Year", sql: Query.count_sql("portal_clazzes", "classes", :year)},
+    "new_activity_counts_by_month" => %Query{name: "New Activity Counts By Month", sql: Query.count_sql("external_activities", "activities", :month)},
+    "new_activity_counts_by_year" => %Query{name: "New Activity Counts By Year", sql: Query.count_sql("external_activities", "activities", :year)},
+    "new_assignment_counts_by_month" => %Query{name: "New Assignment Counts By Month", sql: Query.count_sql("portal_offerings", "assignments", :month)},
+    "new_assignment_counts_by_year" => %Query{name: "New Assignment Counts By Year", sql: Query.count_sql("portal_offerings", "assignments", :year)},
+  }
+
+  @impl true
+  def mount(_params, _session, socket) do
+    socket = socket
```

Comment:
I think `socket` here is what you were calling `state` in the home page live view.
Not sure why `socket` is used here?

Reply:
The socket is different from the state.  The socket is passed to use by the liveview router plug on each page load and is a struct that is analogous to the `conn` struct for regular requests that contains connection information.  

You can see all the elements of the socket struct here:

https://github.com/phoenixframework/phoenix_live_view/blob/main/lib/phoenix_live_view/socket.ex#L62-L73

within that struct there is an `assigns` map that is used in the same manner as the state in a genserver - it holds the current liveview state and anything in that map is available in the templates using the `@` prefix.